src/test/java/io/sealights/plugins/sealightsjenkins/utils/StringUtilsTest.java

```diff
@@ -74,4 +77,53 @@ public void trimStartByValue_lengthLowerThenWantedLength_shouldReturnOriginalStr
         //Assert
         Assert.assertEquals(original,trimmed);
     }
+
+    @Test
+    public void convertKeyValueStringToMap_multipleParams_shouldReturnMap(){
+        //Arrange
+        String original = "param1:val1$$param2:val2";
+        Map<String, String> expected = new HashMap<>();
+        expected.put("param1", "val1");
+        expected.put("param2", "val2");
+
+        //Act
+        Map<String, String> actual = StringUtils.convertKeyValueStringToMap(original);
+
+        //Assert
+        Assert.assertEquals(actual, expected);
+
+
+    }
+
+    @Test
+    public void convertKeyValueStringToMap_multipleColon_shouldSplitByFirst(){
+        //Arrange
+        String original = "url:http://some.com";
```

Comment:
What happens if the URL contains port?
Please add test.
http://www.fake-url.com:8080/?a=b